pkg/cmd/client.go

```diff
@@ -18,7 +24,70 @@ type ClientFlowsConfig struct {
 	Flow      string `usage:"Flow name" env:"KNOW_FLOW"`
 }
 
+func (s *Client) getClientFromArchive(archive string) (client.Client, error) {
+	// unpack to tempdir
+	tmpDir, err := os.MkdirTemp(os.TempDir(), "knowledge-retrieve-*")
+	if err != nil {
+		return nil, err
+	}
+	defer os.RemoveAll(tmpDir)
+
+	r, err := zip.OpenReader(archive)
+	if err != nil {
+		return nil, err
+	}
+	defer r.Close()
+
+	if len(r.File) != 2 {
+		return nil, fmt.Errorf("knowledge archive must contain exactly two files, found %d", len(r.File))
+	}
+
+	dbFile := ""
+	vectorStoreFile := ""
+	for _, f := range r.File {
+		rc, err := f.Open()
+		if err != nil {
+			return nil, err
+		}
+		defer rc.Close()
+
+		path := filepath.Join(tmpDir, f.Name)
+		if f.FileInfo().IsDir() {
+			continue
+		}
+
+		f, err := os.OpenFile(path, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, f.Mode())
+		if err != nil {
+			return nil, err
+		}
+		defer f.Close()
+
+		if _, err := io.Copy(f, rc); err != nil {
+			return nil, err
+		}
+		_ = f.Close()
```

Comment:
Looks like f and rc shoud be closed with defer so no longer need to close it here?

Reply:
I'm closing them right there to prevent possible resource leaks while `defer`'ing within the for-loop (pretty unlikely with only two expected files, but still :grimacing: